sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/security/authorisation/AuthorisationService.java

```diff
@@ -20,24 +29,46 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(AuthorisationService.class);
 
     private final SecurityContextUtil securityContextUtil;
+    private final SiteTreeUtil siteTreeUtil;
 
     private final PractitionerServiceClient practitionerServiceClient;
     private final RoleServiceClient roleServiceClient;
+    private final SiteServiceClient siteServiceClient;
 
     @Autowired
     public AuthorisationService(final SecurityContextUtil securityContextUtil,
+                                final SiteTreeUtil siteTreeUtil,
                                 final PractitionerServiceClient practitionerServiceClient,
-                                final RoleServiceClient roleServiceClient) {
+                                final RoleServiceClient roleServiceClient,
+                                final SiteServiceClient siteServiceClient) {
         this.securityContextUtil = securityContextUtil;
+        this.siteTreeUtil = siteTreeUtil;
         this.practitionerServiceClient = practitionerServiceClient;
         this.roleServiceClient = roleServiceClient;
+        this.siteServiceClient = siteServiceClient;
     }
 
+    public boolean authorise(String requiredPermission, List<?> sites, String methodName) {
```

Comment:
These are all public methods that would be used across the application. Please add javadoc for all explaining the purpose and parameters.

About methodName:
1. Perhaps rename it since it's related to get site id from list?
2. Add a comment to explain why is it a string and used via reflection